[BUG] Cannot Deploy MLeap flavor of spark model in Sagemaker

### Willingness to contribute

Yes. I would be willing to contribute a fix for this bug with guidance from the MLflow community.

### MLflow version

1.27.0

### System information

- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Linux Ubuntu 18.04 (mlflow-pyfunc:1.27.0)
- **Python version**: 3.9.5

Notes: 
- Have Tried also with mlflow-pyfunc:1.26.1 container (and client libs) and encounter the same issue, this is the first time attempting to save to MLeap format.
- Container is pushed from Windows 10 machine (not sure if that matters but thought it wouldn't hurt to add)
- Model is built in Databricks ML 11.1 Beta Runtime (for mlflow 1.26.1 version, Databricks ML 11.0 Runtime was used)



### Describe the problem
I am trying to save my spark model in mleap format and deploy my mleap model to a sagemaker endpoint.  I save and push the model over to sagemaker model registry, and subsequently another script is used to create an endpoint variant (boto3 sagemaker api) using the model that was pushed over using the mlflow.sagemaker API

The endpoint spits out some error logs from the scoring server (Java) and fails to deploy with the following message:

_Exception in thread "main" org.mlflow.sagemaker.PredictorLoadingException: Failed to load the configuration for the MLflow model at the specified path._

### Tracking information

_No response_

### Code to reproduce issue
Create mlflow-pyfunc:1.27.0 container:
```shell
mlflow build-and-push-container
```

In Databricks Notebook:
```python
from pyspark.ml import Pipeline

"""
Construct Model Pipeline using pyspark.ml.feature(s)
Training schema: {'text' -> String, 'label' -> Int}
"""

model = pipeline.fit(training)
sample_model_input = spark.createDataFrame(training.select('text').head(10))
mlflow.spark.log_model(model, 'mleap_model', sample_input=sample_model_input )
```

Once the model has been registered in the Databricks tracking server (Through the Run UI), the following is ran:
```python
import mlflow.sagemaker

model_name = "my-model"
model_version = 1

sm_model_name = f"{model_name}-v{model_version}"
mlflow_model_uri = f"models:/{model_name}/{model_version}"

exec_role_arn = "<EXEC-ROLE-ARN>"
bucket = "<MY-S3-BUCKET>"
sm_image_url = "<ACCOUNT-ECR>/mlflow-pyfunc:1.27.0"
aws_region = "us-east-2"

mlflow.sagemaker.push_model_to_sagemaker(sm_model_name, mlflow_model_uri, execution_role_arn=exec_role_arn,
                                             bucket=bucket, image_url=sm_image_url, region_name=aws_region, flavor='mleap')
```

The boto3 sagemaker client API is then used to deploy the saved mleap model

### Other info / logs

Stack Trace from cloudwatch logs:
```java-traceback
2022/07/22 19:21:57 INFO mlflow.models.container: Got sigterm signal, exiting.
Exception in thread "main" org.mlflow.sagemaker.PredictorLoadingException: Failed to load the configuration for the MLflow model at the specified path.
  #011at org.mlflow.sagemaker.ScoringServer.loadPredictorFromPath(ScoringServer.java:79)
  #011at org.mlflow.sagemaker.ScoringServer.<init>(ScoringServer.java:70)
  #011at org.mlflow.sagemaker.ScoringServer.main(ScoringServer.java:258)
Caused by: com.fasterxml.jackson.databind.exc.UnrecognizedPropertyException: Unrecognized field "databricks_runtime" (class org.mlflow.models.Model), not marked as ignorable (9 known properties: "utc_time_created", "artifact_path", "flavors", "mlflow_version", "rootPath", "signature", "run_id", "model_uuid", "input_example"]) at [Source: (File); line: 2, column: 44] (through reference chain: org.mlflow.models.Model["databricks_runtime"])
  #011at com.fasterxml.jackson.databind.exc.UnrecognizedPropertyException.from(UnrecognizedPropertyException.java:61)
  #011at com.fasterxml.jackson.databind.DeserializationContext.handleUnknownProperty(DeserializationContext.java:840)
  #011at com.fasterxml.jackson.databind.deser.std.StdDeserializer.handleUnknownProperty(StdDeserializer.java:1179)
  #011at com.fasterxml.jackson.databind.deser.BeanDeserializerBase.handleUnknownProperty(BeanDeserializerBase.java:1592)
  #011at com.fasterxml.jackson.databind.deser.BeanDeserializerBase.handleUnknownVanilla(BeanDeserializerBase.java:1570)
  #011at com.fasterxml.jackson.databind.deser.BeanDeserializer.vanillaDeserialize(BeanDeserializer.java:294)
  #011at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:151)
  #011at com.fasterxml.jackson.databind.ObjectMapper._readMapAndClose(ObjectMapper.java:4202)
  #011at com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:3070)
  #011at org.mlflow.utils.SerializationUtils.parseFromFile(SerializationUtils.java:81)
  #011at org.mlflow.utils.SerializationUtils.parseYamlFromFile(SerializationUtils.java:76)
  #011at org.mlflow.models.Model.fromConfigPath(Model.java:70)
  #011at org.mlflow.models.Model.fromRootPath(Model.java:60)
  #011at org.mlflow.sagemaker.ScoringServer.loadPredictorFromPath(ScoringServer.java:76)
  #011... 2 more
```

### What component(s) does this bug affect?

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [x] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/pipelines`: Pipelines, Pipeline APIs, Pipeline configs, Pipeline Templates
- [ ] `area/projects`: MLproject format, project running backends
- [X] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

### What interface(s) does this bug affect?

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

### What language(s) does this bug affect?

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

### What integration(s) does this bug affect?

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [X] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

--- a/mlflow/java/scoring/src/main/java/org/mlflow/models/Model.java
+++ b/mlflow/java/scoring/src/main/java/org/mlflow/models/Model.java
@@ -48,6 +48,9 @@ public class Model {
   @JsonProperty("mlflow_version")
   private String mlflowVersion;
 
+  @JsonProperty("databricks_runtime")
+  private String databricksRuntime;
+
   private String rootPath;
 
   /**
@@ -98,6 +101,14 @@ public class Model {
   /** @return The version of MLflow with which the model was saved */
   public Optional<String> getMlflowVersion() {
     return Optional.ofNullable(this.mlflowVersion);
+  }
+
+  /**
+   * @return If the model was trained on Databricks, the version the Databricks Runtime
+   * that was used to train the model
+   */
+  public Optional<String> getDatabricksRuntime() {
+    return Optional.ofNullable(this.databricksRuntime);
   }
 
   /** @return The path to the root directory of the MLflow model */

--- a/mlflow/java/scoring/src/test/java/org/mlflow/ModelTest.java
+++ b/mlflow/java/scoring/src/test/java/org/mlflow/ModelTest.java
@@ -50,6 +50,18 @@ public class ModelTest {
   }
 
   @Test
+  public void testModelIsLoadedCorrectlyWhenDatabricksRuntimeExists() {
+    String configPath = getClass().getResource("sample_model_root/MLmodel.with.databricks_runtime").getFile();
+    try {
+      Model model = Model.fromConfigPath(configPath);
+      Assert.assertTrue(model.getDatabricksRuntime().isPresent());
+    } catch (IOException e) {
+      e.printStackTrace();
+      Assert.fail("Encountered an exception while reading the model from a configuration path!");
+    }
+  }
+
+  @Test
   public void testModelIsLoadedFromYamlUsingRootPathCorrectly() {
     String rootPath = getClass().getResource("sample_model_root").getFile();
     try {

--- a/mlflow/java/scoring/src/test/resources/org/mlflow/models/sample_model_root/MLmodel.with.databricks_runtime
+++ b/mlflow/java/scoring/src/test/resources/org/mlflow/models/sample_model_root/MLmodel.with.databricks_runtime
@@ -1,0 +1,23 @@
+flavors:
+  mleap:
+    mleap_version: 0.8.1
+    model_data: mleap/model
+utc_time_created: '2018-08-10 18:34:28.720095'
+run_id: c228016dea284522882b657d91eeffa6
+artifact_path: model
+model_uuid: 4ab08bf9d91e4535bc2719f9210840c3
+mlflow_version: 1.25.0
+databricks_runtime: databricks_runtime: 11.0-cpu-ml-scala2.12
+signature:
+  inputs: '[{"name": "fixed acidity", "type": "double"}, {"name": "volatile acidity",
+    "type": "double"}, {"name": "citric acid", "type": "double"}, {"name": "residual
+    sugar", "type": "double"}, {"name": "chlorides", "type": "double"}, {"name": "free
+    sulfur dioxide", "type": "double"}, {"name": "total sulfur dioxide", "type": "double"},
+    {"name": "density", "type": "double"}, {"name": "pH", "type": "double"}, {"name":
+    "sulphates", "type": "double"}, {"name": "alcohol", "type": "double"}]'
+  outputs: '[{"name": null, "type": "double"}]'
+input_example: {
+  "type": "dataframe",
+  "format": "split",
+  "artifact_path": "input_example.json"
+}

--- a/mlflow/java/scoring/src/test/resources/org/mlflow/models/sample_model_root/MLmodel.with.databricks_runtime
+++ b/mlflow/java/scoring/src/test/resources/org/mlflow/models/sample_model_root/MLmodel.with.databricks_runtime
@@ -7,7 +7,7 @@ run_id: c228016dea284522882b657d91eeffa6
 artifact_path: model
 model_uuid: 4ab08bf9d91e4535bc2719f9210840c3
 mlflow_version: 1.25.0
-databricks_runtime: databricks_runtime: 11.0-cpu-ml-scala2.12
+databricks_runtime: 11.0-cpu-ml-scala2.12
 signature:
   inputs: '[{"name": "fixed acidity", "type": "double"}, {"name": "volatile acidity",
     "type": "double"}, {"name": "citric acid", "type": "double"}, {"name": "residual

